### PR TITLE
BUG: Fix for issue #12553

### LINKED ISSUE #12553: SeriesGroupby.nunique raises an IndexError on empty Series
#### Code Sample, a copy-pastable example if possible

``` python
In [18]: b = pandas.Series()

In [19]: g = b.groupby(level = 0)

In [20]: g.nunique()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-20-fbbfc3108eac> in <module>()
----> 1 g.nunique()

/usr/local/lib/python2.7/dist-packages/pandas/core/groupby.pyc in nunique(self, dropna)
   2693 
   2694         out = np.add.reduceat(inc, idx).astype('int64', copy=False)
-> 2695         return Series(out if ids[0] != -1 else out[1:],
   2696                       index=self.grouper.result_index,
   2697                       name=self.name)

IndexError: index 0 is out of bounds for axis 0 with size 0
```
#### Expected Output

``` python
0
```
#### output of `pd.show_versions()`

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.6.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-79-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.1
pip: 8.0.2
setuptools: 20.1.1
Cython: None
numpy: 1.10.4
scipy: 0.13.3
statsmodels: 0.5.0
IPython: 4.1.1
sphinx: None
patsy: 0.2.1
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: None
tables: 3.1.1
numexpr: 2.2.2
matplotlib: 1.3.1
openpyxl: 1.7.0
xlrd: 0.9.2
xlwt: 0.7.5
xlsxwriter: None
lxml: None
bs4: 4.2.1
html5lib: 0.999
httplib2: 0.8
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
Jinja2: None
```

### PR DESCRIPTION
**Pull request by thanasis2028**
closes #12553

### CODE REVIEW

**Comment by jreback** [comment]
thanks. See the contributing guide [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

Like to have tests!

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -2790,6 +2790,10 @@ def true_and_notnull(x, *args, **kwargs):
     def nunique(self, dropna=True):
         """ Returns number of unique elements in the group """
         ids, _, _ = self.grouper.group_info
+        
+        if len(ids) == 0:  # bufix for 12553
+            return Series([])
```
Comment (MaximilianR) [inline pandas/core/groupby.py]: This should just be `self._constructor`, so objects that inherit can keep their type

File: pandas/core/groupby.py [hunk 2]
```diff
@@ -2820,7 +2825,7 @@ def nunique(self, dropna=True):
             inc[idx] = 1
 
         out = np.add.reduceat(inc, idx).astype('int64', copy=False)
-        res = out if ids[0] != -1 else out[1:]
+        res = out if ids[0] != -1 else out[1:])
```
Comment (MaximilianR) [inline pandas/core/groupby.py]: typo?

File: pandas/core/groupby.py (hunk 1 above)
Comment (thanasis2028) [inline pandas/core/groupby.py]: Using `self._constructor` causes an error for me. And the already existing `return` returns a Series() anyway. I don't understand the reason to change this.

File: pandas/core/groupby.py (hunk 2 above)
Comment (thanasis2028) [inline pandas/core/groupby.py]: sry fixed.

**Comment by jreback** [comment]
in the future, no need to open a NEW pull-request, just force push to the old one.

**Pull request closed without merging**